### PR TITLE
Issue 150 lift bin dir

### PR DESCRIPTION
**Pull request by pgoodman**

### CODE REVIEW

File: remill/Arch/Runtime/Operators.h [hunk 1]
```diff
@@ -1185,6 +1185,14 @@ T _ZeroVec(void) {
       __asm__ __volatile__ ("" ::: "memory"); \
     } while (false)
 
+// A 'compiler' barrier that forced a particular value to be used in a specific
+// spot.
+#define BarrierUsedHere(x) \
```
Comment (mike-myers-tob) [inline remill/Arch/Runtime/Operators.h]: I'm not sure I understand these compiler barriers. `BarrierReorder` prevents the compiler from reordering of instructions before and after it, but does its effect extend to a certain scope around it? If `BarrierReorder` prevents a reordering, shouldn't `BarrierUsedHere` prevent a value from being used (it seems to do the opposite: force it to be used)?

I think what is needed is a one-or-two line comment for each Barrier operator that explains its purpose: when is it needed? Is it to trick the compiler or the direct the CPU?

Nitpick: esthetically -> aesthetically

File: tests/AArch64/Run.cpp [hunk 2]
```diff
@@ -300,6 +300,7 @@ static void RunWithFlags(const test::TestInfo *info,
   // native program state recorded before executing the native testcase,
   // but after swapping execution to operate on `gStack`.
   if (!sigsetjmp(gJmpBuf, true)) {
+    std::fesetenv(FE_DFL_ENV);
```
Comment (mike-myers-tob) [inline tests/AArch64/Run.cpp]: Does this require `#pragma STDC FENV_ACCESS ON`?

Do we want to warn if `std::fesetenv()` returns a non-zero value, as in it failed to control the FP environment?

File: remill/Arch/Runtime/Operators.h (hunk 1 above)
Comment (pgoodman) [reply remill/Arch/Runtime/Operators.h]: I will document these in my next PR.

File: tests/AArch64/Run.cpp (hunk 2 above)
Comment (pgoodman) [reply tests/AArch64/Run.cpp]: Clang doesn't support the pragma, and libc correctly implements the function. It's more that LLVM doesn't yet understand that these functions mutate the floating point environment, and therefore affect optimizations of nearby floating point code.

**Pull request merged**